drivers/include/net/netdev.h

```diff
@@ -54,6 +54,7 @@ extern "C" {
 #include <sys/uio.h>
 
 #include "net/netstats.h"
+#include "net/l2filter.h"
```

Comment:
>If a clash occurs with a header from our codebase the header is flawed IMHO, not the consumer of that header (in the same codebase). I would rather take a look into the header and assure the clash does not happen there (e.g. with proper prefixing according to our naming conventions), and not in the consumers of that header.

Prefixing according to our coding conventions surely helps, but does not resolve the problem of clashing with external packages. You yourself gave a good example for what I am talking about.

Nevertheless, the question is not **how** to design a good API, but **whether** a header file should be included that is definitly not used. There are no arguments against `ifdefing` unused header files, besides the *bloating*. But, IMO, bloating is a good price to pay to reduce
1. global name space pollution
2. compilation time
3. strange behaviours due to compiler hints, like e.g. pragmas
4. potential clash of pre-processor statements
5. .. and depending on the order or a misconfigured build system, external software, where we do not have full control (besides of patching), might be affected by above bullet points.

I see that there is not really a point reasoning here as the discussion will spin around forever. I am just surprised that the harms of unused includes are not considered as a bad style and that flaws (IMO) are blindly copied.

@miri64 go ahead and merge.